### DISCUSSION TITLE
Partial key matching in setMutationDefaults

### DISCUSSION THREAD

**Discussion by KamyarP** [TanStack/query#3474] [Q&A] [answered]
Is there any way that I can specify part of the mutation keys in `setMutationDefaults`?
I have defined my keys like below (simplified):
```
export const commentKeys = {
    all: ['comments'] as const,
    mutations: (postId: string) => [...commentKeys.all, postId] as const,
    forPost: (postId: string) => [...commentKeys.all, postId] as const
};
```
Now whenever the users goes back online, I'd like to have all mutations starting with `comments` regardless of the postId so I can persist them on server.
I have included the postId in my mutation keys, so I can call `cancelQueries` for each key in my default `mutationFn` to avoid clashes.

**Comment by TkDodo** [comment, marked as answer]
Yes, the defaults also allow for partial matching, so:

```
queryClient.setMutationDefaults(commentKeys.all, options)
```

will set defaults for all keys starting with `['comments']`

**Reply by KamyarP** [reply]
Perfect! Is this true for `cancelQueries` too? And is there any doc mentioning methods with partial support?

**Reply by TkDodo** [reply]
yes. everything that takes a `filter` argument has partial support, as `{ exact: boolean }` is part of the filter. For everything that affects multiple things, like `cancelQueries`, `invalidateQueries`, `getQueriesData` etc (with the plural queries in the name), `exact` defaults to `false`, meaning it matches fuzzily. Exceptions are methods that are geared towards single queries, like `getQueryData` and `setQueryData`, where `exact` defaults to `true` (note the single `query` in the name).

**Reply by thiagobrez** [reply]
Hey @TkDodo , is there a reason why `queryClient.setMutationDefaults(['SOME_KEY', <someUUID>], options)` shouldn't work? It doesn't throw errors or anything, it just don't seem to set the defaults I want. This worked properly before I introduced the UUID in the mutation key though.

I'm using react-query `3.24.4`

Only thing related I found is this: https://github.com/TanStack/query/issues/2983, in v4

Thanks in advance!

**Reply by TkDodo** [reply]
it will set the defaults for mutations that use this exact uuid only. show a codesandbox reproduction please.
